gc.c

```diff
@@ -9659,6 +9679,18 @@ gc_start_internal(rb_execution_context_t *ec, VALUE self, VALUE full_mark, VALUE
     return Qnil;
 }
 
+void
+rb_gc_prepare_heap(void)
+{
+   /* Promote everyone as candidate for old gen, but don't let the GC run
+    * while we iterate over the heap. */
+   VALUE gc_disabled = rb_gc_disable_no_rest();
```

Comment:
Is there a possibility of GC running during iteration? `gc_set_candidate_object_i` can't possibly trigger GC and it doesn't look like `rb_objspace_each_objects` can trigger GC during iteration.

Reply:
Yeah, I think that's a leftover from the transient heap.